src/protocol/taiko_alethia/TaikoInbox.sol

```diff
@@ -42,32 +48,36 @@ contract TaikoInbox {
             require(lookahead.isCurrentPreconfer(msg.sender), "not current preconfer");
         }
 
-        bytes[] memory attributes = new bytes[](3);
         uint256 _lastPublicationId = lastPublicationId;
 
         // Build the attribute for the anchor transaction inputs
-        require(anchorBlockId >= block.number - maxAnchorBlockIdOffset, "anchorBlockId is too old");
-        bytes32 anchorBlockhash = blockhash(anchorBlockId);
-        require(anchorBlockhash != 0, "blockhash not found");
-        attributes[ANCHOR_TX] = abi.encode(anchorBlockId, anchorBlockhash);
+        require(anchorBlockId >= block.number - maxAnchorBlockIdOffset, "anchorBlockId too old");
 
-        // Build the attribute to link back to the previous publication Id;
-        attributes[PREV_PUBLICATION] = abi.encode(_lastPublicationId);
+        Metadata memory metadata;
+        metadata.anchorBlockId = anchorBlockId;
+        metadata.anchorBlockHash = blockhash(anchorBlockId);
+        require(metadata.anchorBlockHash != 0, "blockhash not found");
+
+        bytes[] memory attributes = new bytes[](3);
+        attributes[METADATA] = abi.encode(metadata);
+        attributes[LAST_PUBLICATION] = abi.encode(_lastPublicationId);
         attributes[BLOB_REFERENCE] = abi.encode(blobRefRegistry.getRef(_buildBlobIndices(nBlobs)));
+
         _lastPublicationId = dataFeed.publish(attributes).id;
 
         // Publish each delayed inclusion as a separate publication
-        IBlobRefRegistry.BlobRef[] memory blobRefs =
-            delayedInclusionStore.processDelayedInclusionByDeadline(block.timestamp);
+        IDelayedInclusionStore.Inclusion[] memory inclusions = delayedInclusionStore.processDueInclusions();
+        uint256 nInclusions = inclusions.length;
+        metadata.isDelayedInclusion = true;
+        for (uint256 i; i < nInclusions; ++i) {
+            attributes[METADATA] = abi.encode(metadata);
+            attributes[LAST_PUBLICATION] = abi.encode(_lastPublicationId);
+            attributes[BLOB_REFERENCE] = abi.encode(inclusions[i]);
```

Comment:
Does it matter that this is a different type to the usual `BLOB_REFERENCE`?

A normal `BLOB_REFERENCE` attribute is the abi encoding of a `BlobRef` object. Within this loop, the reference is the abi encoding of an `Inclusion` object (which contains the BlobRef hash, not the full BlobRef).

Perhaps it's fine because the `isDelayedInclusion` flag can be used to recognise how to interpret it.

Reply:
I think it should be fine. It will be safe to assume in the future the forced inclusion data will use a different format/schema than the regular block proposal blobs.